File: docs/guides/ruby/savePopularRecords.rb
require "algolia"

begin
  client = Algolia::SearchClient.create("ALGOLIA_APPLICATION_ID", "ALGOLIA_API_KEY")

  records = []

  client.browse_objects(
    index_name = "<YOUR_INDEX_NAME>"
  ) do |resp|
    resp.hits.each { |hit|
      record = hit
      record["isPopular"] = hit["nbFollowers"] > 1000000
      records.append(record)
    }
  end

  client.save_objects("<YOUR_INDEX_NAME>", records)
rescue Exception => e
  puts("An error occurred: #{e}")
end

File: docs/guides/php/src/savePopularRecords.php
<?php

require __DIR__.'/../vendor/autoload.php';
use Algolia\AlgoliaSearch\Api\SearchClient;

try {
    $client = SearchClient::create('ALGOLIA_APPLICATION_ID', 'ALGOLIA_API_KEY');

    $results = $client->browseObjects('<YOUR_INDEX_NAME>');

    $records = [];
    foreach ($results as $hit) {
        $records[] = [
            'twitterHandle' => $hit['twitterHandle'],
            'nbFollowers' => $hit['nbFollowers'],
            'isPopular' => $hit['nbFollowers'] > 1000000,
        ];
    }

    $client->saveObjects(
        '<YOUR_INDEX_NAME>',
        $records,
    );
} catch (Exception $e) {
    echo $e->getMessage().PHP_EOL;
}
